packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -304,6 +295,58 @@ class LambdaFunctionHotswapOperation implements HotswapOperation {
     // run all of our updates in parallel
     return Promise.all(operations);
   }
+
+  /**
+   * After a Lambda Function is updated, it cannot be updated again until the
+   * `State=Active` and the `LastUpdateStatus=Successful`.
+   *
+   * Depending on the configuration of the Lambda Function this could happen relatively quickly
+   * or very slowly. For example, Zip based functions _not_ in a VPC can take ~1 second whereas VPC
+   * or Container functions can take ~25 seconds (and 'idle' VPC functions can take minutes).
+   */
+  private async waitForLastUpdateStatusComplete(currentFunctionConfiguration: AWS.Lambda.FunctionConfiguration, lambda: AWS.Lambda): Promise<void> {
+    let delay = 1; // in seconds
+    // if the function is deployed in a VPC or if it is a container image function
+    // then the update will take much longer and we can wait longer between checks
+    if (currentFunctionConfiguration.VpcConfig || currentFunctionConfiguration.PackageType === 'Image') {
+      delay = 5;
+    }
+
+    // configure a custom waiter to wait for the function update to complete
+    (lambda as any).api.waiters.updateFunctionCodeToFinish = {
+      name: 'UpdateFunctionCodeToFinish',
+      operation: 'getFunction',
+      // equates to 1 minute for zip function not in a VPC and
+      // 5 minutes for container functions or function in a VPC
+      maxAttempts: 60,
+      delay,
+      acceptors: [
+        {
+          state: 'success',
+          matcher: 'path',
+          argument: "(Configuration.LastUpdateStatus == 'Successful' && Configuration.State == 'Active')",
+          expected: true,
+        },
+        {
+          state: 'failure',
+          matcher: 'path',
+          argument: 'Configuration.LastUpdateStatus',
+          expected: 'Failed',
+        },
+        {
+          state: 'retry',
+          matcher: 'path',
+          argument: 'Configuration.LastUpdateStatus',
+          expected: 'InProgress',
+        },
```

Comment:
Do we actually need this? I thought `'retry'` was the default if the response was neither `'success'` nor `'failure'`?

Reply:
I guess we don't. I was following the logic [here](https://github.com/aws/aws-sdk-js/blob/1c987aac65ead3afce626b5264f759b62f2585f0/apis/lambda-2015-03-31.waiters2.json#L47), but I actually think it is more clear without specifying `retry` so I went ahead and changed it.